libraries/chain/include/eosio/chain/backing_store/kv_context_rocksdb.hpp

```diff
@@ -0,0 +1,451 @@
+#warning replace with make_shared_bytes
+      const uint32_t final_value_size = backing_store::payer_in_value_size + value_size;
+      auto result = eosio::session::shared_bytes(final_value_size);
+
+      char buf[backing_store::payer_in_value_size];
+      memcpy(buf, &payer, backing_store::payer_in_value_size);
+      std::memcpy(result.data(), buf, backing_store::payer_in_value_size);
+      std::memcpy(result.data() + backing_store::payer_in_value_size, value, value_size);
+      return result;
+   }
+
+
+   static inline eosio::session::shared_bytes make_prefix_key(uint64_t contract, const char* user_prefix, uint32_t user_prefix_size) {
+      static auto rocks_prefix = make_rocksdb_contract_kv_prefix();
+
+      auto result = eosio::session::shared_bytes(rocks_prefix.size() + sizeof(contract) + user_prefix_size);
+      std::memcpy(result.data(), rocks_prefix.data(), rocks_prefix.size());
+      b1::chain_kv::insert_key(result, rocks_prefix.size(), contract);
+      std::memcpy(result.data() + rocks_prefix.size() + sizeof(contract), user_prefix, user_prefix_size);
+      return result;
+   }
+
+   static inline eosio::session::shared_bytes
+   make_composite_key(uint64_t contract, const char* user_prefix, uint32_t user_prefix_size, const char* key, uint32_t key_size) {
+      static auto rocks_prefix = make_rocksdb_contract_kv_prefix(); 
+
+      auto result = eosio::session::shared_bytes(rocks_prefix.size() + sizeof(contract) + user_prefix_size + key_size);
+      std::memcpy(result.data(), rocks_prefix.data(), rocks_prefix.size());
+      b1::chain_kv::insert_key(result, rocks_prefix.size(), contract);
+      std::memcpy(result.data() + rocks_prefix.size() + sizeof(contract), user_prefix, user_prefix_size);
+      std::memcpy(result.data() + rocks_prefix.size() + sizeof(contract) + user_prefix_size, key, key_size);
+      return result;
+   }
+
+   static inline eosio::session::shared_bytes make_composite_key(const eosio::session::shared_bytes& prefix,
+                                                                 const char* key, uint32_t key_size) {
+      auto result = eosio::session::shared_bytes(prefix.size() + key_size);
+      std::memcpy(result.data(), prefix.data(), prefix.size());
+      std::memcpy(result.data() + prefix.size(), key, key_size);
+      return result;
+   }
+
+   static inline int32_t compare_bytes(const eosio::session::shared_bytes& left, const eosio::session::shared_bytes& right) {
+      if (left < right) {
+          return -1;
+      }
+
+      if (left > right) {
+          return 1;
+      }
+
+      return 0;
+   };
+
+   template <typename Session>
+   struct kv_iterator_rocksdb : kv_iterator {
+      using session_type = std::remove_pointer_t<Session>;
+
+      uint32_t&                       num_iterators;
+      uint64_t                        kv_contract{0};
+      const char*                     kv_user_prefix{nullptr};
+      uint32_t                        kv_user_prefix_size{0};
```

Comment:
`kv_user_prefix` & `kv_user_prefix_size` are not used.

Reply:
removed